# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/DOMMediaStream.cpp
# Commit: 10d327bb5893
# Full Hash: 10d327bb5893211f030222b02456bfc11e81a943
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1705080
# File Overlap Count: 1
# Description:
#   Bug 1705080, make MediaStreams behave as normal EventTargets, r=jib
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115269
# ==============================================================================

diff -r 802480d6b25b -r 10d327bb5893 dom/media/DOMMediaStream.cpp
--- a/dom/media/DOMMediaStream.cpp	Tue May 18 16:19:44 2021 +0000
+++ b/dom/media/DOMMediaStream.cpp	Tue May 18 16:21:27 2021 +0000
@@ -97,7 +97,6 @@
 NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN_INHERITED(DOMMediaStream,
                                                 DOMEventTargetHelper)
   tmp->Destroy();
-  NS_IMPL_CYCLE_COLLECTION_UNLINK(mWindow)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mTracks)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mConsumersToKeepAlive)
   NS_IMPL_CYCLE_COLLECTION_UNLINK_WEAK_PTR
@@ -105,7 +104,6 @@
 
 NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN_INHERITED(DOMMediaStream,
                                                   DOMEventTargetHelper)
-  NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mWindow)
   NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mTracks)
   NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mConsumersToKeepAlive)
 NS_IMPL_CYCLE_COLLECTION_TRAVERSE_END
@@ -118,7 +116,7 @@
 NS_INTERFACE_MAP_END_INHERITING(DOMEventTargetHelper)
 
 DOMMediaStream::DOMMediaStream(nsPIDOMWindowInner* aWindow)
-    : mWindow(aWindow),
+    : DOMEventTargetHelper(aWindow),
       mPlaybackTrackListener(MakeAndAddRef<PlaybackTrackListener>(this)) {
   nsresult rv;
   nsCOMPtr<nsIUUIDGenerator> uuidgen =
@@ -350,7 +348,7 @@
 }
 
 already_AddRefed<DOMMediaStream> DOMMediaStream::Clone() {
-  auto newStream = MakeRefPtr<DOMMediaStream>(GetParentObject());
+  auto newStream = MakeRefPtr<DOMMediaStream>(GetOwner());
 
   LOG(LogLevel::Info,
       ("DOMMediaStream %p created clone %p", this, newStream.get()));
@@ -403,7 +401,7 @@
 
 already_AddRefed<nsIPrincipal> DOMMediaStream::GetPrincipal() {
   nsCOMPtr<nsIPrincipal> principal =
-      nsGlobalWindowInner::Cast(mWindow)->GetPrincipal();
+      nsGlobalWindowInner::Cast(GetOwner())->GetPrincipal();
   for (const auto& t : mTracks) {
     if (t->Ended()) {
       continue;